src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -529,13 +531,30 @@ SyntaxKind.GreaterThanToken or
                     SyntaxKind.GreaterThanEqualsToken;
         }
 
-        private static PostSkipAction SkipBadPatternListTokens<T>(
+        private static PostSkipAction SkipBadPatternListTokens_Normal<T>(
+            LanguageParser @this, ref SyntaxToken open, SeparatedSyntaxListBuilder<T> list, SyntaxKind expectedKind, SyntaxKind closeKind)
+            where T : CSharpSyntaxNode
+        {
+            return SkipBadPatternListTokens_DoNotCallDirectly(@this, ref open, list, expectedKind, closeKind, exitOnColon: false);
+        }
+
+        private static PostSkipAction SkipBadPatternListTokens_ExitOnColon<T>(
             LanguageParser @this, ref SyntaxToken open, SeparatedSyntaxListBuilder<T> list, SyntaxKind expectedKind, SyntaxKind closeKind)
             where T : CSharpSyntaxNode
+        {
+            return SkipBadPatternListTokens_DoNotCallDirectly(@this, ref open, list, expectedKind, closeKind, exitOnColon: true);
+        }
+
+        private static PostSkipAction SkipBadPatternListTokens_DoNotCallDirectly<T>(
+            LanguageParser @this, ref SyntaxToken open, SeparatedSyntaxListBuilder<T> list, SyntaxKind expectedKind, SyntaxKind closeKind, bool exitOnColon)
+            where T : CSharpSyntaxNode
         {
             if (@this.CurrentToken.Kind is SyntaxKind.CloseParenToken or SyntaxKind.CloseBraceToken or SyntaxKind.CloseBracketToken or SyntaxKind.SemicolonToken)
                 return PostSkipAction.Abort;
 
+            if (exitOnColon && @this.CurrentToken.Kind is SyntaxKind.ColonToken)
+                return PostSkipAction.Abort;
```

Comment:
don't pass in exitOnColon.  Just infer that rom the term-state and doc here why.